### DISCUSSION TITLE
Josh Goldberg - typescript-eslint

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#197] [Announcements]
Andrew: Hello and welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin.

## Topics

- TSEslint v6
- TypeScript syntax best practices
- async code best practices
- what's formatting vs style
- now to set up static analysis tooling
- dprint
- https://typescript-eslint.io/blog/announcing-typescript-eslint-v6/

## Questions

- [x] Have you done anything fun with type systems lately?
- [x] What the difference between formatting and style? What are some concrete examples?
- [x] What are some of your typescript best practices ?
- [x] I hear you have opinions on async code too? Are those opinions encoded in rules?
- [x] How have the default configurations changed. What should I use?
- [x] Have have any new rules been added?
- [x] What are your thoughts about eslint ability to cache?
- [x] What's your spiciest dev take?
- [x] What planned for tseslint v7?

**Comment by hipstersmoothie** [comment]
https://github.com/fabian-hiller/valibot
https://kysely.dev/

**Comment by JoshuaKGoldberg** [comment]
* [Knip](https://github.com/webpro/knip)
* [Nonviolent Communication](https://en.wikipedia.org/wiki/Nonviolent_Communication)

**Comment by just-be-dev** [comment]
- https://github.com/qewer33/qpaperOS
- https://github.com/denosaurs/deno_python
